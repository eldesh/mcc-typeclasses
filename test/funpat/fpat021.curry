-- Since function patterns are effectively transformed into constraint
-- guards, their arguments cannot introduce any dynamic instances, which
-- is similar to lazy patterns. However, in contrast to lazy patterns,
-- the corresponding instances must be present in the environment so that
-- the program can construct the constraint expression at runtime. (See
-- also fpat019.curry and fpat020.curry for variants of this example.)

data Numeric a = Num a => Numeric a

f (id (Numeric x)) = True
g (Numeric x `const` _) = True
