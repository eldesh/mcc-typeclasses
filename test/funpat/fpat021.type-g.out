Num _1 => Numeric _1 -> Bool
